src/arfsdao.ts

```diff
@@ -568,8 +813,336 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		});
 		return drive.build();
 	}
+
+	async getAllFoldersOfPublicDrive(driveId: DriveID): Promise<ArFSPublicFolder[]> {
+		let cursor = '';
+		let hasNextPage = true;
+		const allFolders: ArFSPublicFolder[] = [];
+
+		while (hasNextPage) {
+			const gqlQuery = buildQuery(
+				[
+					{ name: 'Drive-Id', value: driveId },
+					{ name: 'Entity-Type', value: 'folder' }
+				],
+				cursor
+			);
+
+			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+			const { data } = response.data;
+			const { transactions } = data;
+			const { edges } = transactions;
+			hasNextPage = transactions.pageInfo.hasNextPage;
+
+			const folders: Promise<ArFSPublicFolder>[] = edges.map(async (edge: GQLEdgeInterface) => {
+				// Iterate through each tag and pull out each drive ID as well the drives privacy status
+				const folderBuilder = new ArFSPublicFolderBuilder();
+				const { node } = edge;
+				const { tags } = node;
+				cursor = edge.cursor;
+				tags.map((tag: GQLTagInterface) => {
+					const key = tag.name;
+					const { value } = tag;
+					switch (key) {
+						case 'App-Name':
+							folderBuilder.appName = value;
+							break;
+						case 'App-Version':
+							folderBuilder.appVersion = value;
+							break;
+						case 'ArFS':
+							folderBuilder.arFS = value;
+							break;
+						case 'Content-Type':
+							folderBuilder.contentType = value as ContentType;
+							break;
+						case 'Drive-Id':
+							folderBuilder.driveId = value;
+							break;
+						case 'Entity-Type':
+							folderBuilder.entityType = value as EntityType;
+							break;
+						case 'Unix-Time':
+							folderBuilder.unixTime = +value;
+							break;
+						case 'Parent-Folder-Id':
+							folderBuilder.parentFolderId = value;
+							break;
+						case 'Folder-Id':
+							folderBuilder.entityId = value;
+							break;
+						default:
+							break;
+					}
+				});
+
+				// Get the drives transaction ID
+				folderBuilder.txId = node.id;
+
+				const txData = await this.arweave.transactions.getData(folderBuilder.txId, { decode: true });
+				const dataString = await Utf8ArrayToStr(txData);
+				const dataJSON = await JSON.parse(dataString);
+
+				// Get the drive name and root folder id
+				folderBuilder.name = dataJSON.name;
+				return folderBuilder.build();
+			});
+			allFolders.push(...(await Promise.all(folders)));
+		}
+		return allFolders;
+	}
+
+	async getPrivateFolder(folderId: FolderID): Promise<ArFSPrivateFolder> {
+		const gqlQuery = buildQuery([
+			{ name: 'Folder-Id', value: folderId },
+			{ name: 'Entity-Type', value: 'folder' },
```

Comment:
Folders do not contain a `Drive-Privacy` tag, this query will always return empty.

Reply:
Fixed: 728c9251caf6ed81e28e010b00acd9d5601c19f9